src/Microsoft.Azure.WebJobs.Host/Settings/ConfigurationSettingProvider.cs

```diff
@@ -0,0 +1,28 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Globalization;
+
+namespace Microsoft.Azure.WebJobs.Host.Settings
+{
+    /// <summary>
+    /// Implements settings provider based on Configuration (app settings, environment variables or command-line params)
+    /// </summary>
+    public class ConfigurationSettingProvider : ISettingProvider
+    {
+        /// <inheritdoc />
+        public bool? Disabled(string methodName)
+        {
+            var disabledSetting = ConfigurationUtility.GetSetting(string.Format(CultureInfo.InvariantCulture, "{0}.Disabled", methodName));
+            if (!string.IsNullOrEmpty(disabledSetting))
+            {
+                return string.Compare(disabledSetting, "false", StringComparison.OrdinalIgnoreCase) != 0;
```

Comment:
could we remove the `IsNullOrEmpty` and return the `Compare` expression for 'true'?

Reply:
we should be even stricter (since that would still allow many different values for false, like 'false', 'disabled', 'stopped', 'please_dont_run_this', etc)

---
In reply to: [144928487](https://github.com/Azure/azure-webjobs-sdk/pull/1379#discussion_r144928487) [](ancestors = 144928487)